### DISCUSSION TITLE
Seperation of session protocol and desktop app?

### DISCUSSION THREAD

**Discussion by peepo5** [oxen-io/session-desktop#2364] [General]
It seems that the session protocol is build directly into the desktop app (as far as I can tell). This may stunt molecularity because everything using the protocol would have to derive the integration from the desktop app, which would be hard to keep up to date and which could also introduce new security issues. If I am correct about the protocol not being an individual component rather than directly integrated, would it be possible to separate the protocol from the interface (like matrix)? Thanks.

**Comment by KeeJef** [comment]
We have started https://github.com/oxen-io/libsession-util/ which is aiming to split out the core protocol of Session, in a way which can be imported into Session-Desktop, Session-Android and Session-iOS such that core features maintain a standard implementation across all platforms
